### DISCUSSION TITLE
MauiIcons suport .NET8

### DISCUSSION THREAD

**Discussion by JohnStavrou** [AathifMahir/MauiIcons#77] [Q&A] [answered]
I've implemented the MauiIcons as instructed, but is seems to throw the follwing error: 
Cannot resolve type "http://www.aathifmahir.com/dotnet/2022/maui/icons:mi:MauiIcon"
Is it maybe because I am running .net8?

Funny enough if i comment out this part of the code, run the app, and then uncomment, it works fine :P
It is a build issue..

**Comment by AathifMahir** [comment, marked as answer]
> I've implemented the MauiIcons as instructed, but is seems to throw the follwing error: 
> Cannot resolve type "http://www.aathifmahir.com/dotnet/2022/maui/icons:mi:MauiIcon"
> Is it maybe because I am running .net8?
> 
> Funny enough if i comment out this part of the code, run the app, and then uncomment, it works fine :P
> It is a build issue..	

Seems like url namespace issue of Maui is not fixed yet.
https://github.com/dotnet/maui/issues/7503

As a workaround, you can new up the Maui icon instance in code behind.

Here's how you can new up the Maui icons,

In the constructor of mainpage.xaml.cs or app.xaml.cs, you can add discarded instance of Maui icon just like below example

```csharp
public MainPage()
    {
        InitializeComponent();
        // Temporary Workaround for url styled namespace in xaml
        _ = new MauiIcon();
    }
```

Also, Thanks for reporting this issue, will update readme with this workaround

**Reply by r1di** [reply]
that fixed it for me, ty(im on .NET 9 RC1)

**Reply by AathifMahir** [reply]
> that fixed it for me, ty(im on .NET 9 RC1)

That's awesome to hear, will further test this with Dotnet 9 once it's GA and remove workaround from the docs
